Parquet/src/main/java/io/deephaven/parquet/ColumnChunkReaderImpl.java

```diff
@@ -175,7 +173,27 @@ public ColumnPageReader next() {
                 currentOffset = file.position() + pageHeader.getCompressed_page_size();
                 remainingValues -= pageHeader.isSetData_page_header() ? pageHeader.getData_page_header().num_values :
                         pageHeader.getData_page_header_v2().getNum_values();
-                return new ColumnPageReaderImpl(channelsProvider, file.position(), pageHeader, decompressor, path, dictionary, getPath(), -1, fieldTypes);
+                final org.apache.parquet.format.Encoding encoding;
+                switch (pageHeader.type) {
+                    case DATA_PAGE:
+                        encoding = pageHeader.getData_page_header().getEncoding();
+                        break;
+                    case DATA_PAGE_V2:
+                        encoding = pageHeader.getData_page_header_v2().getEncoding();
+                        break;
+                    default:
+                        throw new UncheckedDeephavenException("Unknown parquet data page header type " + pageHeader.type);
+                }
+                final Dictionary pageDictionary =
+                        (encoding == org.apache.parquet.format.Encoding.PLAIN_DICTIONARY
+                                || encoding == org.apache.parquet.format.Encoding.RLE_DICTIONARY)
+                        ? dictionary
```

Comment:
Is there any value in renaming `dictionary` to `columnChunkDictionary`, in light of what we know now?

Reply:
Mmm... I don't think so.  In terms of what it may apply to, parquet dictionaries are per row group / column chunk, so there is no ambiguity there once we understand how parquet dictionaries work.  In terms of when they do apply, we are out of luck even with that name because now we know that column chunk dictionaries are, in  term of pages "maybe I am your dictionary, I mean, if you have a dictionary, then it is me, but maybe you don't have one".  :-/.  Which is why I think pageDictionary helps.  Maybe as a concept pageDictionary is not great btw.  It would be better to have a boolean "does the dictionary apply to me".  But that would make the change bigger (we would have to change the API for ToDictionaryPage and recruit ToPage to support it somehow, without looking too weird for the other types...  It may be a better change tho... but I think I am punting, your call.

Reply:
I think the `pageDictionary` concept is just fine, and I don't see any need to do the boolean-oriented refactoring you described. I think this PR should merge without further changes.